### PR TITLE
virtio_port: fix python3 format issues during data transfer

### PR DESCRIPTION
**Pull request by vivianQizhu**
id: 1625835
Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@xutian I think that's not really proper in these scenarios, most `data` `recv` variants will added different content later, while the added data is bytes, so here use `b""` instead of `""`. That's meaningless to define a `null_byte` for them.

File: shared/scripts/virtio_console_guest.py [hunk 1]
```diff
@@ -419,7 +418,7 @@ def _reconnect_none_mode(self):
             #       when bz796048 is resolved.
             data = ""
```
Comment (balamuruhans) [inline shared/scripts/virtio_console_guest.py]: can be removed ?

Comment (vivianQizhu) [reply shared/scripts/virtio_console_guest.py]: @balamuruhans Right.

**Comment by vivianQizhu** [comment]
@balamuruhans Updated, please help review, thanks.

File: shared/scripts/virtio_console_guest.py [hunk 2]
```diff
@@ -353,9 +353,8 @@ def _none_mode(self):
             """
             Read and write to device in blocking mode
             """
-            data = ""
             while not self.exit_thread.isSet():
-                data = ""
+                data = b""
                 for desc in self.in_files:
                     data += os.read(desc, self.cachesize)
                 if data != "":
```
Comment (balamuruhans) [inline shared/scripts/virtio_console_guest.py]: we should consider the change here.

**Comment by vivianQizhu** [comment]
@balamuruhans Thanks, updated.

**Comment by vivianQizhu** [comment]
@luckyh Updated. Please help review, thanks.
python2:
```
char value and type: ('\xe7', <type 'str'>)
_char value and type: ('\xe7', <type 'str'>)
```
ord('\xe7') = 231
python3:
```
char value and type: 74 <class 'int'>
_char value and type: b'J' <class 'bytes'>
```
ord(b'J') = 74
I don't find other way besides `ord()` to handle the python3 conversion, so use `char == _char or char == ord(_char)` to adapt to python 2 and python3

**Comment by vivianQizhu** [comment]
Test passed on both RHEL7 and RHEL8:
```
1-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.my-provider.virtio_console.spread_linear.virtserialport.migration.offline/debug.log:01:16:46 INFO | PASS 1-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.my-provider.virtio_console.spread_linear.virtserialport.migration.offline
2-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.my-provider.virtio_console.spread_linear.virtserialport.interrupted_transfer.stop_cont.long/debug.log:01:21:37 INFO | PASS 2-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.my-provider.virtio_console.spread_linear.virtserialport.interrupted_transfer.stop_cont.long
```
```
1-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.myprovider.virtio_console.spread_linear.virtserialport.migration.offline/debug.log:01:16:06 INFO | PASS 1-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.myprovider.virtio_console.spread_linear.virtserialport.migration.offline
2-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.myprovider.virtio_console.spread_linear.virtserialport.interrupted_transfer.stop_cont.long/debug.log:01:20:27 INFO | PASS 2-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.myprovider.virtio_console.spread_linear.virtserialport.interrupted_transfer.stop_cont.long
```

**Comment by vivianQizhu** [comment]
Besides, for `char = b"%c" % random.randrange(0, 255)`, only when `random.randrange(0, 255)` < 128 `char.decode()` works, otherwise `UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte`. So decode() can not solve this issue.

File: virttest/qemu_virtio_port.py [hunk 3]
```diff
@@ -789,12 +789,12 @@ def run_normal(self):
                     # Compare the received data with the control data
                     for char in buf:
                         _char = self.buff.popleft()
-                        if char == _char:
+                        if char == _char or char == ord(_char):
```
Comment (luckyh) [inline virttest/qemu_virtio_port.py]: Using `bytearray` and `struct.pack()` would make an unified approach to process the check.

For example:
```python
for c_code in bytearray(buf):
    char = struct.pack('B', c_code)
    _char = self.buff.popleft()
    if char == _char:
        ...
```

File: virttest/qemu_virtio_port.py [hunk 4]
```diff
@@ -126,7 +126,7 @@ def clean_port(self):
         self.open()
         ret = select.select([self.sock], [], [], 1.0)
         if ret[0]:
-            buf = self.sock.recv(1024)
+            buf = self.sock.recv(1024).decode()
```
Comment (luckyh) [inline virttest/qemu_virtio_port.py]: Suggest:
```python
buf = self.sock.recv(1024)
logging.debug("Rest in socket: " + repr(buf))
```

File: virttest/qemu_virtio_port.py (hunk 3 above)
Comment (vivianQizhu) [reply virttest/qemu_virtio_port.py]: @luckyh Great, it works always. I will update it this way. Thanks.

**Pull request merged**